config/deploy/docker/lib/command_args.rb

```diff
@@ -1,13 +1,45 @@
 # frozen_string_literal: true
 
 require 'yaml'
+require 'dotenv'
 
 class CommandArgs
   attr_accessor :deployments
 
   def initialize
+    Dotenv.load('.env', '.env.local')
+
     path = Pathname.new(__FILE__).join('..', '..', 'assets', 'secret.deploy.values.yml')
-    config = YAML.load_file(path)
+
+    if File.exist?(path)
+      local_config = YAML.load_file(path)
+      local_config = nil if local_config.empty?
+    else
+      local_config = nil
+    end
+
+    remote_config_text = AwsSdkHelpers::Helpers.get_secret(ENV['SECRETS_YML_SECRET_ARN'])
+    remote_config = YAML.safe_load(remote_config_text, [Symbol], aliases: true)
+
+    if !local_config.nil? && local_config != remote_config
+      puts 'Local secrets.yml differs from remote config, would you like to pull down the remote version? This will overwrite your local file. [y/N]'
+      unsure = $stdin.readline
+      if unsure.chomp.downcase.match?(/y(es)?/)
+        File.write(path, remote_config_text)
+        config = remote_config
+      else
+        tmppath = "/tmp/remote-secrets-#{Time.now.to_i}.yml"
+        File.write(tmppath, remote_config_text)
+        puts "Okay. Remote config saved to #{tmppath} for your convenience. (Local is at #{path})"
+        exit
+      end
+    elsif remote_config.nil?
+      config = local_config
+      puts "[WARN] ❗ Remote secrets.yml not found, using local: #{path}"
+    else
+      config = remote_config
+    end
```

Comment:
@andyzito , is there a problem if local differs from remote?  I often comment out some lines (like a few in the staging stanza) so I can call `bin/vault_deploy staging` and just run the ones that remain.    I just had a deploy get through the first 5 or so, but fail for random TCP related connections on the 6th.  Then when I commented out the first 5 it refused to run.  Can we maybe download the remote one, show a diff of the two and ask if you'd like to continue with the local file?   Or is there a reason it needs to use the remote one?